src/SeriesProvider/SeriesProvider.py

```
@@ -70,18 +70,188 @@ def request_input(self, seriesDescription: SeriesDescription, timeDescription: T
-        log(f'''Request input failure, 
-                    Reason: {validated_merged_result.nonCompleteReason} \n 
+        # If neither were valid then we attempt to interpolate
+        log(f'Init Interpolation from {seriesDescription}|{timeDescription}')
+        interpolation_results = self.__interpolate_series(validated_merged_result)
+        validated_interpolation_results = self.__generate_resulting_series(seriesDescription, timeDescription, interpolation_results.data)
+        
+        return validated_interpolation_results
+      
+    
+    def __interpolate_series(self, inSeries: Series) -> Series: 
+         """This method will interpolate the results from the query if the gaps between the NaNs are not larger than the limit
+
+        Args:
+            inSeries (Series): The incomplete merged result of the DB and DI queries
+
+        Returns:
+            Series : The Series with new interpolated Inputs added
+        """
+         timeDescription = inSeries.timeDescription
+         seriesDescription = inSeries.description
+         interpolationParameters = seriesDescription.interpolationParameters
+
+         # If there is only one Input (only one data point) then we do not interpolate
+         if(len(inSeries.data) <= 1):
+             log(f'''Interpolation error,
+                    Reason: Only one Input found in series.\n 
                     {seriesDescription} \n 
                     {timeDescription} \n 
-                    Series Storage Result: {series_storage_results.data} \n 
-                    Data Ingestion Result: {None if data_ingestion_results == None else data_ingestion_results.data} \n
-                    Merged Result: {validated_merged_result.data}
-            ''')
-        return validated_merged_result
+                ''')
+             return inSeries
 
-            
+         if interpolationParameters is None:
```

This should hard fail, it should raise an exception.